src/renderTopLanguages.js

```diff
@@ -72,6 +98,26 @@ const renderTopLanguages = (topLangs, options = {}) => {
   if (hide_title) {
     height -= 30;
   }
+
+  if(layout === 'compact') {
+    return `
+      <svg width="${width+50}" height="${height}" viewBox="0 0 ${width+50} ${height}" fill="none" xmlns="http://www.w3.org/2000/svg">
+        <style>
+          .header { font: 600 18px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${titleColor} }
+          .lang-name { font: 400 14px 'Segoe UI', Ubuntu, Sans-Serif; fill: ${textColor} }
+        </style>
+        <rect width="${width+50}" height="${height}" fill='#fff' stroke="#E4E2E2" />
+        ${hide_title ? '' : `<text x="25" y="35" fill='#333' class='header'>Top Languages</text>`}
+        <svg x='25' y='${hide_title ? 25 : 50}'>
+          ${langs.map(lang => `<rect width='${((lang.size / totalSize) * width).toFixed(2)}' height='10' fill='${lang.color}' rx='5'/>`).join('')}
+        </svg>
+        <svg x="25" y="${hide_title ? 50 : 75}">
+          ${createLanguageTextNode({langs, totalSize})}
+        </svg>
+      </svg>
+    `;
+  }
+
```

Comment:
We don't have to build return a whole new SVG string just to change the layout, this approach would cause duplicate logic/code in `title (hide_title), border (hide_border), svg width, also the CSS.`

So it would be better if we just conditionally render the inner body itself instead of recreating the whole svg again.

Reply:
Yes, @anuraghazra I completely agree with you. I'm a little bit confused SVG and this is the first time I'm creating SVG.

Reply:
Yeah we just need to conditionally render this part

```svg
        <svg x='25' y='${hide_title ? 25 : 50}'>
          ${langs.map(lang => `<rect width='${((lang.size / totalSize) * width).toFixed(2)}' height='10' fill='${lang.color}' rx='5'/>`).join('')}
        </svg>
        <svg x="25" y="${hide_title ? 50 : 75}">
          ${createLanguageTextNode({langs, totalSize})}
        </svg>
```

Reply:
Hey @anuraghazra, I'm done with changes. Could you please take a look at PR?

Demo:

![top langs with layout compact](https://github-readme-stats-two-rho.vercel.app/api/top-langs?username=sindresorhus&layout=compact)

![top langs with layout compact](https://github-readme-stats-two-rho.vercel.app/api/top-langs?username=sindresorhus&layout=compact&theme=dark)

![top langs with layout compact](https://github-readme-stats-two-rho.vercel.app/api/top-langs?username=sindresorhus&layout=compact&theme=dracula)

![top langs with layout compact](https://github-readme-stats-two-rho.vercel.app/api/top-langs?username=sindresorhus&layout=compact&theme=tokyonight&hide_title=true)

Reply:
Looks good, just need to give the whole progressbar some border radius, i know we cannot do it per language <rect> because it's looking weird. so i think best thing to do would be to utilize svg masks

Reply:
I tried to apply some border-radius (`rx`) but it looks weird. I don't know about SVG masks. Do you've any resource for understanding the SVG mask?

Reply:
Hey @anuraghazra, I tried hard to apply `border-radius` on the left side of the first progressive bar and right side of the last progressive bar but failed to do so. I found that we can also do the same thing using paths.

```html
<path d="M 0 20 h 100 q 4 0 4 4 v 0 q 0 4 -4 4 h -100 z" fill="#333"/>
```